tests/test_technology.py

```diff
@@ -0,0 +1,285 @@
+        _validate_unit("darkmatter", "fuel")
+
+
+def test_validate_quantity():
+    assert _validate_quantity(power_unyt, 'power') == 10 * (MW)
+    assert _validate_quantity(energy_unyt, 'energy') == 10 * (MW * hr)
+    assert _validate_quantity(time_unyt, 'time') == 1 * (hr)
+    assert _validate_quantity("10 Horsepower**-1",
+                              'spec_power') == 10 * (Horsepower**-1)
+    assert _validate_quantity(10 * (Horsepower * day)**-1,
+                              'spec_energy') == 10 * ((Horsepower * day)**-1)
+
+    with pytest.raises(TypeError) as e:
+        _validate_quantity(10 * MW, "energy")
+    with pytest.raises(UnitParseError) as e:
+        _validate_quantity("10 darkmatter", "energy")
+
+    with pytest.raises(KeyError) as e:
+        _validate_quantity("10 darkmatter", "fuel")
+
+
+def test_initialize():
+    assert PLANET_EXPRESS.technology_name == TECH_NAME
+    assert PLANET_EXPRESS.technology_type == 'base'
+    assert PLANET_EXPRESS.capacity == 0.0
+    assert PLANET_EXPRESS.capital_cost == 0.0
+    assert PLANET_EXPRESS.om_cost_fixed == 0.0
+    assert PLANET_EXPRESS.om_cost_variable == 0.0
+    assert PLANET_EXPRESS.fuel_cost == 0.0
+    assert PLANET_EXPRESS.unit_power == MW
+    assert PLANET_EXPRESS.unit_time == hr
+    assert PLANET_EXPRESS.unit_energy == MW * hr
+
+
+def test_attribute_types():
+    assert isinstance(PLANET_EXPRESS.capacity, unyt.array.unyt_quantity)
+    assert isinstance(PLANET_EXPRESS.capital_cost, unyt.array.unyt_quantity)
+    assert isinstance(PLANET_EXPRESS.om_cost_fixed, unyt.array.unyt_quantity)
+    assert isinstance(PLANET_EXPRESS.om_cost_variable, unyt.array.unyt_quantity)
+    assert isinstance(PLANET_EXPRESS.fuel_cost, unyt.array.unyt_quantity)
+    assert isinstance(PLANET_EXPRESS.unit_power, unyt.unit_object.Unit)
+    assert isinstance(PLANET_EXPRESS.unit_energy, unyt.unit_object.Unit)
+    assert isinstance(PLANET_EXPRESS.unit_time, unyt.unit_object.Unit)
+
+
+def test_capacity():
+    with pytest.raises(ValueError) as e:
+        PLANET_EXPRESS.capacity = dict_type
+    with pytest.raises(UnitParseError) as e:
+        PLANET_EXPRESS.capacity = unknown_str
+    with pytest.raises(AssertionError) as e:
+        PLANET_EXPRESS.capacity = energy_str
+
+    PLANET_EXPRESS.capacity = power_unyt
+    assert PLANET_EXPRESS.capacity.value == 10.0
+    assert PLANET_EXPRESS.capacity.units == MW
+
+    PLANET_EXPRESS.capacity = power_str
+    assert PLANET_EXPRESS.capacity.value == 10.0
+    assert PLANET_EXPRESS.capacity.units == MW
+
+    PLANET_EXPRESS.capacity = int_val
+    assert PLANET_EXPRESS.capacity.value == 10
+    assert PLANET_EXPRESS.capacity.units == MW
+
+    PLANET_EXPRESS.capacity = str_val
+    assert PLANET_EXPRESS.capacity.value == 10.0
+    assert PLANET_EXPRESS.capacity.units == MW
+
+    PLANET_EXPRESS.capacity = float_val * other_power_unyt
+    assert PLANET_EXPRESS.capacity.value == pytest.approx(0.007457, 0.005)
```

Comment:
Probably not necessary, but you could define the HP variables up above instead so they can be written as `HP-variable-name` when you do the assert statements.

Reply:
You mean the `0.007457`? I suppose I could, but I think that might make it less clear. You can make a suggestion if you'd like!